app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java

```diff
@@ -340,23 +341,44 @@ private void startDecodeQrCodeImages(List<Uri> uris) {
         QrDecodeTask task = new QrDecodeTask(this, (results) -> {
             List<CharSequence> errors = new ArrayList<>();
             List<VaultEntry> entries = new ArrayList<>();
+            List<GoogleAuthInfo.Export> googleAuthExports = new ArrayList<>();
+
             for (QrDecodeTask.Result res : results) {
                 if (res.getException() != null) {
                     errors.add(buildImportError(res.getFileName(), res.getException()));
                     continue;
                 }
 
                 try {
-                    GoogleAuthInfo info = GoogleAuthInfo.parseUri(res.getResult().getText());
-                    VaultEntry entry = new VaultEntry(info);
-                    entries.add(entry);
-                } catch (GoogleAuthInfoException e) {
+                    Uri scanned = Uri.parse(res.getResult().getText());
+                    if (Objects.equals(GoogleAuthInfo.getUriScheme(scanned), GoogleAuthInfo.SCHEME_EXPORT)) {
```

Comment:
I don't think the ``GoogleAuthInfo.getUriScheme`` function is needed. Wouldn't ``Objects.equals(scanned.getScheme(), GoogleAuthInfo.SCHEME_EXPORT)`` work just the same?